src/ussd_popi_rapidpro.js

```diff
@@ -993,77 +1142,38 @@ go.app = function() {
-                    "you stop getting msgs?"
-                ),
-                error: $(
-                    "Sorry we don't recognise that reply. Please enter the number next to your " +
-                    "answer."
-                ),
-                choices: [
-                    new Choice("yes", $("Yes")),
-                    new Choice("no", $("No"))
-                ],
-                next: "state_submit_opt_out"
-            });
-        });
-
-        self.add("state_delete_data", function(name) {
-            return new MenuState(name, {
-                question: $(
-                    "We hold your info for historical/research/statistical reasons after you " +
-                    "opt out. Do you want to delete your info after you stop getting messages?"
-                ),
-                error: $(
-                    "Sorry we don't recognise that reply. Please enter the number next to your " +
-                    "answer."
-                ),
-                choices: [
-                    new Choice("state_delete_confirm", $("Yes")),
+                    new Choice("state_submit_opt_out", $("Yes")),
                     new Choice("state_submit_opt_out", $("No"))
                 ]
             });
         });
 
-        self.add("state_delete_confirm", function(name) {
-            return new MenuState(name, {
-                question: $(
-                    "All your info will be permanently deleted in the next 7 days. We'll stop " +
-                    "sending you messages. Please select Next to continue:"
-                ),
-                error: $(
-                    "Sorry we don't recognise that reply. Please enter the number next to your " +
-                    "answer."
-                ),
-                choices: [
-                    new Choice("state_submit_opt_out", $("Next"))
-                ]
+        self.add("state_message_unhelpful_or_unknown", function(name) {
+            return new EndState(name, {
+                next: "state_start",
+                text: $("Thank you for supporting MomConnect. You won't get any more messages from us." +
+                        "\n" +
+                        "For any medical concerns, please visit a clinic."
+                )
             });
         });
 
         self.add("state_submit_opt_out", function(name, opts) {
             var msisdn = utils.normalize_msisdn(self.im.user.addr, "ZA");
             var answers = self.im.user.answers;
-            var forget = answers.state_delete_data === "state_delete_confirm";
-            var loss = answers.state_loss_messages === "state_loss_delete_data";
-            var loss_forget = answers.state_loss_delete_data === "yes";
+            var forget = _.toUpper(answers.state_anonymous_data) === "YES";
+            var loss = _.toUpper(answers.state_loss_messages) === "YES";
+            var loss_forget = _.toUpper(answers.state_submit_opt_out) === "YES";
```

Comment:
These are choicestates, the value won't be YES, that is only the label that the user sees, see the lines above you replaced

Reply:
Okay i see